app/scripts/controllers/transactions/index.js

```diff
@@ -377,8 +377,10 @@ class TransactionController extends EventEmitter {
       // add nonce to txParams
       // if txMeta has lastGasPrice then it is a retry at same nonce with higher
       // gas price transaction and their for the nonce should not be calculated
+      const { customNonceValue = null } = txMeta
       const nonce = txMeta.lastGasPrice ? txMeta.txParams.nonce : nonceLock.nextNonce
```

Comment:
It looks like we're using `nonceLock.nextNonce` here and not using it, in the case where `txMeta.lastGasPrice` is falsy and `customNonceValue` is set

Reply:
Actually I think the damage is done up on line 376, not on this line.

Reply:
The `nonceLock` could still be referenced here in the custom nonce case (if `lastGasPrice` is truthy).